metrics-graphite/src/main/java/com/codahale/metrics/graphite/Graphite.java

```diff
@@ -181,6 +181,11 @@ public void close() throws IOException {
         } finally {
             this.socket = null;
         }
+        
+        // if hostname is used, re-resolve DNS in connect() by clearing the address
+        if (this.hostname != null) {
```

Comment:
The `address` thats referring to is the local variable `address`, not the shadowed field `address`. It's confusing. But we never write to the field `address` if the class was initialized with a hostname, so the hostname will always be re-resolved (at least we will attempt to reresolve it, Java's DNS caching could be screwing that up).